kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseSiteService.java

```diff
@@ -1536,6 +1536,19 @@ public String siteReference(String id)
 		return getAccessPoint(true) + Entity.SEPARATOR + id;
 	}
 
+	/**
+	 * @inheritDoc
+	 */
+	public String idFromSiteReference(String ref) {
+
+		String[] resourceParts = ref.split("/");
```

Comment:
If you have a ref you should call EntityManager.newReference(String reference) and then call getId()